cmd/osquery-perf/agent.go

```diff
@@ -2349,6 +2400,16 @@ func main() {
 			tmpl = tmplss[i%len(tmplss)]
 		}
 
+		if tmpl.Name() == "iphone_14.6.tmpl" || tmpl.Name() == "ipad_13.18.tmpl" {
+			model := "iPhone 14,6"
+			if tmpl.Name() == "ipad_13.18.tmpl" {
+				model = "iPad 13,18"
+			}
+			go runAppleIDeviceMDMLoop(i, stats, model, *serverURL, *mdmSCEPChallenge, *mdmCheckInInterval)
+			time.Sleep(sleepTime)
+			continue
+		}
```

Comment:
Just wondering if it wouldn't be better for future maintainability to have this iDevice thing part of the `agent` as for the other hosts, or maybe create a distinct `agent` type and have a constructor function take care of returning the right agent based on the template?

Not a blocker, just that `osquery-perf` is already very "loosely organized" and it is a bit tricky to understand and maintain. Maybe for some future work, to refactor some of this (not just iDevice stuff).